crates/bevy_ecs/src/entity/map_entities.rs

```diff
@@ -27,19 +27,57 @@ use bevy_utils::EntityHashMap;
 /// }
 ///
 /// impl MapEntities for Spring {
-///     fn map_entities(&mut self, entity_mapper: &mut EntityMapper) {
-///         self.a = entity_mapper.get_or_reserve(self.a);
-///         self.b = entity_mapper.get_or_reserve(self.b);
+///     fn map_entities<M: Mapper>(&mut self, entity_mapper: &mut M) {
+///         self.a.map_entities(entity_mapper);
+///         self.b.map_entities(entity_mapper)
 ///     }
 /// }
 /// ```
 ///
 pub trait MapEntities {
     /// Updates all [`Entity`] references stored inside using `entity_mapper`.
     ///
-    /// Implementors should look up any and all [`Entity`] values stored within and
-    /// update them to the mapped values via `entity_mapper`.
-    fn map_entities(&mut self, entity_mapper: &mut EntityMapper);
+    /// Only updates the references for which there is a mapping.
+    fn map_entities<M: Mapper>(&mut self, entity_mapper: &mut M);
```

Comment:
@SkiFire13 I think this works better, there is less repetition overall.

The main remaining problem is that the function requires `entity_mapper: &mut M` even though for some mappers (the `SimpleEntityMapper`), `&M` would be enough

Reply:
I think this is fine, if it ever becomes a problem `Mapper` could be implemented for `&SimpleEntityMapper`.

Reply:
I wanted to do that, but `Mapper` has 
```
/// Map an entity to another entity
fn map(&mut self, entity: Entity) -> Entity;
```
which requires `&mut self`.

So we'd need two `MutMapper` and `Mapper` traits

Reply:
I'm not sure how that's a problem, you would just get `self` to be a `&mut &SimpleEntityMapper` which should be fine.